File: magma-platform/magma-platform-base/src/main/java/de/jalumu/magma/platform/base/config/MagmaConfig.java
package de.jalumu.magma.platform.base.config;

import java.io.File;

public interface MagmaConfig {

    void read();

    void read(File file);

    void setObject(String key, Object o);

    void setString(String key, String s);

    void setInt(String key, int i);

    void save();

    void save(File file);

}

File: magma-platform/magma-platform-base/src/main/java/de/jalumu/magma/platform/base/config/MagmaConfig.kt
package de.jalumu.magma.platform.base.config

import java.io.File

interface MagmaConfig {
    fun read()
    fun read(file: File?)
    fun setObject(key: String?, o: Any?)
    fun setString(key: String?, s: String?)
    fun setInt(key: String?, i: Int)
    fun save()
    fun save(file: File?)
}
